src/neptune/internal/backends/swagger_client_wrapper.py

```diff
@@ -28,66 +29,65 @@
-    PROJECT_NAME_COLLISION = "PROJECT_NAME_COLLISION"
-    PROJECT_KEY_INVALID = "PROJECT_KEY_INVALID"
-    PROJECT_NAME_INVALID = "PROJECT_NAME_INVALID"
-    EXPERIMENT_NOT_FOUND = "EXPERIMENT_NOT_FOUND"
-    AUTHORIZATION_TOKEN_EXPIRED = "AUTHORIZATION_TOKEN_EXPIRED"
-
     def __init__(self, api_method):
         self._api_method = api_method
 
     @staticmethod
     def handle_neptune_http_errors(response, exception: Optional[HTTPError] = None):
+        from neptune.management.exceptions import (
+            IncorrectIdentifierException,
+            ManagementOperationFailure,
+            ObjectNotFound,
+            ProjectKeyCollision,
+            ProjectKeyInvalid,
+            ProjectNameCollision,
+            ProjectPrivacyRestrictedException,
+            ProjectsLimitReached,
+        )
+
         try:
             body = response.json() or dict()
         except Exception:
             body = {}
 
+        error_processors: dict[str, Callable[[dict], ManagementOperationFailure]] = {
+            "ATTRIBUTES_PER_EXPERIMENT_LIMIT_EXCEEDED": lambda response_body: NeptuneFieldCountLimitExceedException(
+                limit=response_body.get("limit", "<unknown limit>"),
+                container_type=response_body.get("experimentType", "object"),
+                identifier=response_body.get("experimentQualifiedName", "<unknown identifier>"),
+            ),
+            "AUTHORIZATION_TOKEN_EXPIRED": lambda _: NeptuneAuthTokenExpired(),
+            "EXPERIMENT_NOT_FOUND": lambda _: ObjectNotFound(),
+            "INCORRECT_IDENTIFIER": lambda response_body: IncorrectIdentifierException(
+                identifier=response_body.get("identifier", "<Unknown identifier>")
+            ),
+            "LIMIT_OF_PROJECTS_REACHED": lambda _: ProjectsLimitReached(),
+            "PROJECT_KEY_COLLISION": lambda response_body: ProjectKeyCollision(
+                key=response_body.get("key", "<unknown key>")
+            ),
+            "PROJECT_KEY_INVALID": lambda response_body: ProjectKeyInvalid(
+                key=response_body.get("key", "<unknown key>"),
+                reason=response_body.get("reason", "Unknown reason"),
+            ),
+            "PROJECT_NAME_COLLISION": lambda response_body: ProjectKeyCollision(
+                key=response_body.get("key", "<unknown key>")
+            ),
+            "PROJECT_NAME_INVALID": lambda response_body: ProjectNameCollision(
+                name=response_body.get("name", "<unknown name>")
+            ),
+            "VISIBILITY_RESTRICTED": lambda response_body: ProjectPrivacyRestrictedException(
+                requested=response_body.get("requestedValue", "the selected"),
+                allowed=response_body.get("allowedValues"),
+            ),
+            "WORKSPACE_IN_READ_ONLY_MODE": lambda response_body: NeptuneLimitExceedException(
+                reason=response_body.get("title", "Unknown reason")
+            ),
+        }
```

Comment:
@Raalsky @BartoszPrusak how do you feel about testing these not-happy paths e2e? Do you think it's worth the effort?

Reply:
In the long term, I would like to add some kind of integration tests that will mimic the possible outcomes of a backend API (mocking responses of `requests`). That would be a much better place for such not-happy paths and e2e could be a bare minimum. Currently -> It's ok but there is no big need for it.

Reply:
I added an e2e test for the error message I introduced, but don't plan on adding more.
I talked to Raalsky and the non-happy paths will be tested in a different environment.